[CSS] Fix serialization issues with clip-path & offset-path

nt1m opened the pull request:
#### 70f48f9012f4066004dc47c186f8f76cfee1a99e
<pre>
[CSS] Fix serialization issues with clip-path &amp; offset-path
<a href="https://bugs.webkit.org/show_bug.cgi?id=262009">https://bugs.webkit.org/show_bug.cgi?id=262009</a>
rdar://115953688

Reviewed by Darin Adler.

Two issues:
- We should serialize in canonical order (&lt;basic-shape&gt; before &lt;geometry-box&gt;)
- `border-box` should be omitted as it is the default &lt;geometry-box&gt; for both properties

This patch also updates the default geometry-box for clip-path to border-box per spec: <a href="https://drafts.fxtf.org/css-masking/#typedef-geometry-box">https://drafts.fxtf.org/css-masking/#typedef-geometry-box</a>

* LayoutTests/fast/masking/parsing-clip-path-shape-expected.txt:
* LayoutTests/fast/masking/parsing-clip-path-shape.html:
* LayoutTests/imported/w3c/web-platform-tests/css/css-masking/parsing/clip-path-valid.html:
* LayoutTests/imported/w3c/web-platform-tests/css/motion/parsing/offset-path-computed-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/motion/parsing/offset-path-parsing-valid-expected.txt:

Rebaseline passing tests.

* LayoutTests/svg/clip-path/clip-path-shape-content-box-expected.svg:
* LayoutTests/svg/clip-path/clip-path-shape-fill-expected.svg:
* LayoutTests/svg/clip-path/clip-path-shape-padding-box-expected.svg:

Update tests to account for the new default geometry box.

* Source/WebCore/css/parser/CSSPropertyParserHelpers.cpp:
(WebCore::CSSPropertyParserHelpers::consumeRayShape):
(WebCore::CSSPropertyParserHelpers::consumeBasicShapeRayOrBox):

Fix parsing.

* Source/WebCore/rendering/RenderElement.cpp:
(WebCore::RenderElement::referenceBoxRect const):
* Source/WebCore/rendering/svg/SVGRenderSupport.cpp:
(WebCore::clipPathReferenceBox):

Fix the default clip-path reference box.

* Source/WebCore/style/StyleBuilderConverter.h:
(WebCore::Style::BuilderConverter::convertPathOperation):
(WebCore::Style::BuilderConverter::convertShapeValue):

Cleanup.

Canonical link: <a href="https://commits.webkit.org/268379@main">https://commits.webkit.org/268379@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/97f15c80382af21d81287297b87d35a9c55a2845

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/19527 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/26/builds/19947 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/20555 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/21418 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/18264 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/19764 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/23/builds/23210 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/16/builds/20089 "Built successfully") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/34/builds/19867 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/19744 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/15/builds/19768 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/16973 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/22274 "Built successfully") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/39/builds/16955 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/17761 "Passed tests") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/1/builds/24072 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/18016 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/17936 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/22044 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/7/builds/18544 "Built successfully") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/17/builds/15714 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/30/builds/17688 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/17596 "Passed tests") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/4669 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/4/builds/22044 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/31/builds/18372 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

darinadler commented on Source/WebCore/css/parser/CSSPropertyParserHelpers.cpp:
```diff
@@ -7161,33 +7161,38 @@ static RefPtr<CSSRayValue> consumeRayShape(CSSParserTokenRange& range, const CSS
+        // FIXME: The current Motion Path spec calls for this to be a <coord-box>, not a <geometry-box>, the difference being that the former does not contain "margin-box" as a valid term.
+        // However, the spec also has a few examples using "margin-box", so there seems to be some abiguity to be resolved. Tracked at https://github.com/w3c/fxtf-drafts/issues/481.
+        if (!box && (box = CSSPropertyParsing::consumeGeometryBox(range)))
+            continue;
```
Typically when I say "current" I like to include a date even though source code control also can show the date and of course the issue also provides a date.

nt1m replied:
Removed "current" from the comment, since it doesn't provide much value here.

nt1m committed:
[CSS] Fix serialization issues with clip-path & offset-path
https://bugs.webkit.org/show_bug.cgi?id=262009
rdar://115953688

Reviewed by Darin Adler.

Two issues:
- We should serialize in canonical order (<basic-shape> before <geometry-box>)
- `border-box` should be omitted as it is the default <geometry-box> for both properties

This patch also updates the default geometry-box for clip-path to border-box per spec: https://drafts.fxtf.org/css-masking/#typedef-geometry-box

* LayoutTests/fast/masking/parsing-clip-path-shape-expected.txt:
* LayoutTests/fast/masking/parsing-clip-path-shape.html:
* LayoutTests/imported/w3c/web-platform-tests/css/css-masking/parsing/clip-path-valid.html:
* LayoutTests/imported/w3c/web-platform-tests/css/motion/parsing/offset-path-computed-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/motion/parsing/offset-path-parsing-valid-expected.txt:

Rebaseline passing tests.

* LayoutTests/svg/clip-path/clip-path-shape-content-box-expected.svg:
* LayoutTests/svg/clip-path/clip-path-shape-fill-expected.svg:
* LayoutTests/svg/clip-path/clip-path-shape-padding-box-expected.svg:

Update tests to account for the new default geometry box.

* Source/WebCore/css/parser/CSSPropertyParserHelpers.cpp:
(WebCore::CSSPropertyParserHelpers::consumeRayShape):
(WebCore::CSSPropertyParserHelpers::consumeBasicShapeRayOrBox):

Fix parsing.

* Source/WebCore/rendering/RenderElement.cpp:
(WebCore::RenderElement::referenceBoxRect const):
* Source/WebCore/rendering/svg/SVGRenderSupport.cpp:
(WebCore::clipPathReferenceBox):

Fix the default clip-path reference box.

* Source/WebCore/style/StyleBuilderConverter.h:
(WebCore::Style::BuilderConverter::convertPathOperation):
(WebCore::Style::BuilderConverter::convertShapeValue):

Cleanup.

Canonical link: https://commits.webkit.org/268379@main
